packages/cli/test/commands/migrate/e2e.ember-app.test.ts

```diff
@@ -0,0 +1,335 @@
+      verifyTsConfig(project.baseDir);
+    });
+
+    test('should have the correct files in the app directory', () => {
+      verifyApp(project.baseDir);
+    });
+
+    test('should have the correct files in the app components directory', () => {
+      verifyAppComponents(project.baseDir);
+    });
+
+    test('should have the correct files in the app services directory', () => {
+      verifyAppServices(project.baseDir);
+    });
+
+    test('should have the correct files in the app templates directory', () => {
+      verifyAppTemplates(project.baseDir);
+    });
+
+    test('should have the correct files in the tests directory', () => {
+      verifyTests(project.baseDir);
+    });
+  });
+
+  describe('ember app with in-repo addon', () => {
+    let project: Project;
+
+    beforeAll(async () => {
+      project = getEmberAppWithInRepoAddonProject();
+      await project.write();
+
+      await runBin('migrate', ['--ci'], {
+        cwd: project.baseDir,
+      });
+    });
+
+    afterAll(() => {
+      project.dispose();
+    });
+
+    test('should have the correct files in the base directory', async () => {
+      verifyBaseDir(project.baseDir);
+      await verifyPackageJson(project.baseDir);
+      verifyReport(project.baseDir);
+      verifyTsConfig(project.baseDir);
+    });
+
+    test('should have the correct files in the app directory', () => {
+      verifyApp(project.baseDir);
+    });
+
+    test('should have the correct files in the app components directory', () => {
+      verifyAppComponents(project.baseDir);
+    });
+
+    test('should have the correct files in the app templates directory', () => {
+      verifyAppTemplates(project.baseDir);
+    });
+
+    test('should have the correct files in the tests directory', () => {
+      verifyTests(project.baseDir);
+    });
+
+    test('should have the correct files in the lib some-addon directory', () => {
+      const someAddonPath = resolve(project.baseDir, 'lib', 'some-addon');
+      const someAddonFiles = readdirSync(someAddonPath);
+      expect(someAddonFiles).toContain('index.js');
+      expect(someAddonFiles).toContain('package.json');
+
+      const innerAddonComponentsPath = resolve(
+        project.baseDir,
+        'lib',
+        'some-addon',
+        'addon',
+        'components'
+      );
+      const innerAddonComponentsFiles = readdirSync(innerAddonComponentsPath);
+      expect(innerAddonComponentsFiles).toContain('greet.ts');
+      expect(
+        readFileSync(resolve(innerAddonComponentsPath, 'greet.ts'), { encoding: 'utf-8' })
+      ).toMatchSnapshot();
+
+      const innerAppComponentsPath = resolve(
+        project.baseDir,
+        'lib',
+        'some-addon',
+        'app',
+        'components'
+      );
+      const innerAppComponentsFiles = readdirSync(innerAppComponentsPath);
+      expect(innerAppComponentsFiles).toContain('greet.js');
+    });
+  });
+
+  describe('ember app with in-repo addon', () => {
```

Comment:
please rename this `ember app with in-repo engine`

Reply:
done